scripts/pdf/generate_pdf.py

```diff
@@ -256,29 +94,195 @@ def main():
                 "15mm",
                 "--margin-bottom",
                 "15mm",
-                *headerFooterCommon,
-                coverFilePath,
+                *header_footer_common,
+                cover_file,
                 "--footer-right",
                 f"Built at {datetime.utcnow().replace(microsecond=0).isoformat()}",
                 "toc",
                 "--xsl-style-sheet",
                 "scripts/pdf/toc-style.xsl",
-                *headerFooterOptions,
-                htmlFilePath,
-                *headerFooterOptions,
-                pdfFilePath,
+                *header_footer_options,
+                html_file,
+                *header_footer_options,
+                pdf_file,
             ]
         )
         output.check_returncode()
 
-    if openPdf:
-        run(["open", pdfFilePath])
+    if args.open_pdf:
+        run(["open", pdf_file])
+
+    mdx_file.unlink()
+    if not args.generate_html_only:
+        html_file.unlink()
+        cover_file.unlink()
+
+
+def setup(args):
```

Comment:
Loved seeing this broken out; makes it a lot easier to see where we'd need to poke if dir structure changed (e.g. if we needed PDFs for unversioned products)